packages/aquedux-client/src/createClient.js

```diff
@@ -0,0 +1,122 @@
+// @flow
+
+import SockJS from 'sockjs-client'
+
+import actionTypes from './constants/actionTypes'
+import configManager, { type AqueduxConfig } from './managers/configManager'
+import { raise, register, unregister, events } from './utils/eventHub'
+import localStorage from './utils/localStorage'
+
+export default function createClient(options: AqueduxConfig) {
+  const { endpoint, hydratedActionTypes } = configManager.setConfig(options)
+
+  let socket: ?SockJS = null
+  let actionStack: Array<Object> = []
+
+  // Ideally, these actions should not be sent back to client. Duplicates definition in middleware
+  function isWiredActionType(actionType: string): boolean {
+    const internalActionTypes = [
+      actionTypes.AQUEDUX_CLIENT_CHANNEL_JOIN,
+      actionTypes.AQUEDUX_CLIENT_CHANNEL_LEAVE,
+      actionTypes.AQUEDUX_CHANNEL_SNAPSHOT
+    ]
+
+    return internalActionTypes.indexOf(actionType) !== -1 || hydratedActionTypes.indexOf(actionType) !== -1
+  }
+
+  function onSend(action) {
+    // If the socket is not open, we stack the actions.
+    if (!socket || socket.readyState !== 1) {
+      actionStack.push(action)
+      return
+    }
+
+    // Append our jwt from the local storage to the action before sending it.
+    const token = localStorage.getItem('aquedux-jwt')
+    if (token !== null) {
+      action = {
+        ...action,
+        token
+      }
+    }
+
+    socket.send(JSON.stringify(action))
+  }
+
+  function onMessage(socket, message) {
+    const json = (() => {
+      try {
+        return JSON.parse(message.data)
+      } catch (e) {
+        return null
+      }
+    })()
+
+    if (!json || !json.type) {
+      console.error(`The client received an invalid message: ${JSON.stringify({ message: json }, null, 2)}`)
+      return
+    }
+
+    // If the message has a jwt, override our own.
+    if (json.token) {
+      localStorage.setItem('aquedux-jwt', json.token)
+    }
+
+    const action = (() => {
+      if (isWiredActionType(json.type)) {
+        /**
+         * We need to change type to bypass potential user middlewares
+         * Real action.type will be restored in aquedux middleware before hitting reducers
+         */
+        return {
+          ...json,
+          originalActionType: json.type,
+          type: actionTypes.AQUEDUX_CLIENT_MESSAGE_RECEIVED
+        }
+      } else {
+        return json
+      }
+    })()
+
+    raise(events.EVENT_ACTION_RECEIVE, action)
+  }
+
+  function onOpen(isRestart = false) {
+    // Try to resubscribe to previously subscribed channels before loosing connection.
+    if (isRestart) {
+      raise(events.EVENT_CHANNEL_RESUBSCRIBE)
```

Comment:
Do we really need this behavior?
Maybe we could resubscribe at each start by default if there is an subscribed channel?
If a user doesn't want this behavior, he can clean/unsubscribe channels before calling `start()` action creator.

I would love to remove `restart` action creator.